lib/ansible/modules/cloud/amazon/s3_bucket.py

```diff
@@ -510,11 +588,13 @@ def main():
             state=dict(default='present', type='str', choices=['present', 'absent']),
             tags=dict(required=False, default=None, type='dict'),
             versioning=dict(default=None, type='bool'),
-            ceph=dict(default='no', type='bool')
+            ceph=dict(default='no', type='bool'),
+            encryption=dict(choices=['none', 'AES256', 'aws:kms']),
+            encryption_key_id=dict()
         )
     )
 
-    module = AnsibleAWSModule(argument_spec=argument_spec)
+    module = AnsibleAWSModule(argument_spec=argument_spec, required_if=[['encryption', 'aws:kms', ['encryption_key_id']]])
```

Comment:
I think we also want to require `encryption` be specified anytime `encryption_key_id` is passed. That way users won't accidentally pass the key_id and assume that it's being used when it may not be.